Show different icons depending on browser 'family', add a fallback for not knowing family

jennifer-shehane opened the pull request:
- Closes #5745 

### User facing changelog

- The browser dropdown selector in the Test Runner now supports showing browser icons other than Chrome.

### Additional details

- Added Firefox icon for firefox family browsers
- Added a 'globe' icon as a default for browsers outside of family `electron`, `chrome` or `firefox`.
- Refreshed the colors to be a bit brighter 😎 

### How has the user experience changed?

#### Before

<img width="118" alt="69180085-b1c75380-0ada-11ea-94ba-a49a6b647910" src="https://user-images.githubusercontent.com/1271364/69714795-832e2600-1135-11ea-83b0-d6738a8bcc8c.png">


#### After

<img width="205" alt="Screen Shot 2019-11-27 at 4 41 40 PM" src="https://user-images.githubusercontent.com/1271364/69714659-364a4f80-1135-11ea-9b59-38999d80791f.png">


### PR Tasks

<!-- 
These tasks must be completed before a PR is merged.
Delete tasks if they are not applicable. 
-->

- [x] Have tests been added/updated?
- [ ] Has the original issue been tagged with a release in ZenHub? <!-- (internal team only)-->
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? <!-- Link to PR here -->
- [NA] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?
- [NA] Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?

jennifer-shehane committed:
have different icons display depending on browser 'family', add a fallback for not knowing family

bahmutov commented:
hmm, I do like colors there ...

brian-mann commented on packages/desktop-gui/src/lib/browser-model.js:
```diff
@@ -25,6 +25,14 @@ export default class Browser {
+      return 'firefox'
+    }
+
+    return 'globe'
```
```
return _
.chain(['chrome', 'firefox'])
.find(_.matches(this.family)
.defaultTo('globe')
.value()
```

flotwig replied:
while we're code-golfing, i might as well throw in my implementation...

```js
({
	'chrome': 'chrome'
	'electron': 'chrome'
	'firefox': 'firefox'
})[this.family] || 'globe'
```

brian-mann commented on packages/desktop-gui/src/app/nav.scss:
```diff
@@ -148,10 +148,11 @@
+  // .chrome > i { color: #1aa15f; }
+  // .canary > i { color: #e4b721; }
+  // .chromium > i { color: #829ac3; }
+  // .electron > i { color: #777; }
```
definitely still want color :D

jennifer-shehane commented:
Ok, I opened this very quickly before I went out of town just to have the code in question fresh somewhere. I was fully expecting to revisit if no one else picked up. :D

jennifer-shehane committed:
Update browsers dropdown

- Add back the colors + make them a little nicer
- Add more believable browsers to config fixture (including Electron)
- Update tests to make them more dynamic to newly added browsers in
config

flotwig commented:
Hmm, the more I look at this, the more I think we should just switch to using the actual, official, rasterized icons for the browsers, not SVG or FontAwesome icons. 

Compare the Firefox icon in Cypress to the real Firefox icon:

![image](https://user-images.githubusercontent.com/1151760/69733908-1ba7b480-10fc-11ea-87f0-e6ab5f082f7f.png)

![image](https://user-images.githubusercontent.com/1151760/69733887-121e4c80-10fc-11ea-8f57-a460805e904f.png)

Or even compare Chrome: 

![image](https://user-images.githubusercontent.com/1151760/69733970-324e0b80-10fc-11ea-9329-18fda74c65f8.png)

![image](https://user-images.githubusercontent.com/1151760/69733942-282c0d00-10fc-11ea-8554-e4c17747b022.png)

Cypress's icon looks like a slightly wrong knockoff in both cases.

I'm assuming the reason why we use the FontAwesome icons is because it's easier, but at this point, we might as well include the actual icons into our product. It will look better and we'll never have to touch this again.

@jennifer-shehane Thoughts?

bahmutov commented on packages/desktop-gui/src/lib/browser-model.js:
```diff
@@ -25,6 +26,14 @@ export default class Browser {
   @computed get icon () {
-    return 'chrome'
+    if (this.family === 'electron') {
+      return 'chrome'
```
it would be nicer to get separate electron icon I think

flotwig replied:
yeah, using their icon would cut down on people confusing electron w/ chrome too: 
![image](https://user-images.githubusercontent.com/1151760/69734815-a5a44d00-10fd-11ea-8422-c45ef6c74578.png)

jennifer-shehane replied:
we're using font-awesome 4.7 here so 😬😬😬

bahmutov commented on packages/desktop-gui/src/lib/browser-model.js:
```diff
@@ -25,6 +26,14 @@ export default class Browser {
+      return 'chrome'
+    }
+
+    return _
```
```js
return _.get({
  chrome: 'chrome',
  firefox: 'firefox'
}, this.family, 'globe') 
```
https://lodash.com/docs/4.17.15#get

jennifer-shehane replied:
lol, everybody has an opinion here 🙃 I agree this is simpler than @brian-mann suggestion.

flotwig replied:
turns out this is implemented in another place too: https://github.com/cypress-io/cypress/blob/6405fa66c688c03bb1188240400db61bc5af42f0/packages/desktop-gui/src/lib/utils.js#L13-L18 https://github.com/cypress-io/cypress/blob/6405fa66c688c03bb1188240400db61bc5af42f0/packages/desktop-gui/src/lib/utils.js#L28-L32

jennifer-shehane commented:
@flotwig oh, I fully agree - the font awesome icons are awful and am just doing this because it's easier.

flotwig commented:
> @flotwig oh, I fully agree - the font awesome icons are awful and am just doing this because it's easier.

Ok cool, want to go ahead and switch it? i think the change is fairly small now that you have this PR

jennifer-shehane commented:
Yes, I won't have time until tomorrow, so if someone wants to pick it up to get it in - also feel free.

jennifer-shehane committed:
Update code to use existing utility for browser icons.

jennifer-shehane committed:
Merge branch 'develop' into issue-5745

jennifer-shehane committed:
Merge branch 'develop' into issue-5745

jennifer-shehane committed:
fix missing closing }

jennifer-shehane committed:
make firefox icon same size as others + fix style that was not being read

jennifer-shehane commented:
I've created a separate issue for using nicer browser icons and will be merging this PR in (in order to prepare for Firefox release). https://github.com/cypress-io/cypress/issues/5921
